napari/plugins/_plugin_manager.py

```diff
@@ -292,6 +293,81 @@ def available_samples(self) -> Tuple[Tuple[str, str], ...]:
             (p, s) for p in self._sample_data for s in self._sample_data[p]
         )
 
+    # THEME DATA ------------------------------------
+
+    def register_theme_colors(
+        self, data: Dict[str, Dict[str, str]], hookimpl: HookImplementation
+    ):
+        """Register theme data dict returned by `napari_experimental_provide_theme`.
+
+        Each key in `data` is a `theme_name` and the value is dictionary of values.
+        """
+        keys = [
```

Comment:
I wonder if we should define this list of keys somewhere in our `utils/theme.py` file and import them here. Seems less likely for something to get out of sync if we were to ever change any of these

Reply:
That makes sense. I've changed the `_themes` dict to use the `EventedDict` and also perform key validation whenever new theme is added. Haven't fully tested it yet so it might change.